### ISSUE TITLE
Trouble building on macOS (10.13.1)

### ISSUE THREAD

**Issue by dzt** [LibreTuner/LibreTuner#2]
After installing all the essential dependencies such as Boost and qt5 I started to get errors for the modules in the `lib` directory so I manually downloaded those repo's from GitHub since the `googletest` and `GSL` folders were empty from their respective links on GitHub and ran `cmake .`

I got the following warnings:

```
CMake Warning at /usr/local/Cellar/cmake/3.7.2/share/cmake/Modules/FindBoost.cmake:761 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.7.2/share/cmake/Modules/FindBoost.cmake:865 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/Cellar/cmake/3.7.2/share/cmake/Modules/FindBoost.cmake:1454 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:10 (find_package)


CMake Warning at /usr/local/Cellar/cmake/3.7.2/share/cmake/Modules/FindBoost.cmake:761 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.7.2/share/cmake/Modules/FindBoost.cmake:865 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/Cellar/cmake/3.7.2/share/cmake/Modules/FindBoost.cmake:1454 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:10 (find_package)
```

Regardless of the warnings, it created a `Makefile` and I ran the `make` command and got the following error:

```
CMake Warning (dev) at copy-headers.cmake:5 (configure_file):
  configure_file called with unknown argument(s):

   Research/LibreTuner/lib/GSL/gsl/ECU

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error: File /Users/peter/Desktop/ECU does not exist.
CMake Error at copy-headers.cmake:5 (configure_file):
  configure_file Problem configuring file


make[2]: *** [lib/GSL/gsl/gsl_inline.h] Error 1
make[1]: *** [lib/GSL/CMakeFiles/copy-headers.dir/all] Error 2
make: *** [all] Error 2
```

Sorry if this issue since a bit confusing, tried by best to give the most detailed description I could. Really appreciate the work you're trying to distribute to the community, I can't wait to get my hands on the codebase and contribute!

Cheers,
Peter

**Comment by jcmnn** [comment]
This repo is slightly outdated because I moved to https://gitlab.com/Altenius/LibreTuner about a month ago. This is supposed to be a mirror but I haven't been updating it. Boost was removed because getting it to work on Windows was too much trouble and the libraries were empty because the submodules were not initialized; I updated the README with instructions on how to download the submodules and have also updated this repo.

Let me know if the issue still persists after updating :).
